src/cdk/table/table.ts

```diff
@@ -452,7 +456,9 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
       // Optional for backwards compatibility, but a view repeater strategy will always
       // be provided.
       @Optional() @Inject(_VIEW_REPEATER_STRATEGY)
-      protected readonly _viewRepeater: _ViewRepeater<T, RenderRow<T>, RowContext<T>>) {
+      protected readonly _viewRepeater: _ViewRepeater<T, RenderRow<T>, RowContext<T>>,
+      @Optional() @Inject(_TABLE_LAYOUT_STRATEGY)
+      private readonly _layoutStrategy: _TableLayoutStrategy|null) {
```

Comment:
While the `|null` here is correct, I believe that ViewEngine will throw an error for it. You might have to make the parameter type non-null, declare a separate nullable property and assign to it:

```
private _layoutStrategy: _TableLayoutStrategy|null;

constructor(@Optional() @Inject(_TABLE_LAYOUT_STRATEGY) layoutStrategy: _TableLayoutStrategy) {
  this._layoutStrategy = layoutStrategy;
}
```

Reply:
Can we make this optional instead? Similar to [this PR](https://github.com/angular/components/pull/20425/files)?

```
private readonly _layoutStrategy?: _TableLayoutStrategy
```

Reply:
Yes, if I remember correctly optional parameters shouldn't throw.